src/BalanceTx.purs

```diff
@@ -295,21 +302,15 @@ balanceTxM (UnbalancedTx { transaction: unbalancedTx, utxoIndex }) = do
                         returnAdaChange ownAddr allUtxos nonAdaBalancedCollTx
                         >>= case _ of
                           Left err -> pure $ Left err
-                          Right unsignedTx ->
-                            signTransaction unsignedTx
+                          Right unsignedTx -> do
+                            signedTx' <- signTransaction unsignedTx
                               <#> note (SignTxError' $ CouldNotSignTx ownAddr)
-
-  -- Some useful spies for debugging.
-  -- case signedTx' of
-  --   Left err -> pure $ Left err
-  --   Right signedTx -> do
-  --     let signedTxBody = (unwrap signedTx).body
-  --         signedTxOuputs = (unwrap signedTxBody).outputs
-  --         x = spy "signedTxInputVal" (getInputValue allUtxos signedTxBody)
-  --         y = spy "signedTxOutputs" signedTxOuputs
-  --         z = spy "signedTxOutputsVal" (Array.foldMap getAmount signedTxOuputs)
-  --         a = spy "signedTxFees" $ (unwrap signedTxBody).fee
-  --     pure $ pure $ signedTx
+                            -- Some post balancer logging:
+                            case signedTx' of
+                              Left err -> pure $ Left err
```

Comment:
This is the 3rd level of Either case split where the Left branch just passes up the error. Can we de-nest it somehow, for example by introducing a local helper function?

Reply:
or the 5th.

Reply:
@vvtran If you wanted to, I don't see why you couldn't use `ExceptT`/`runExceptT` locally in situations where the `Either`s get deeply nested

Reply:
@ngua ah, this could work, perhaps that's what other people have meant and I've misunderstood.